snapshot-labs/sx-ui
# feat: make SendToken/SendNft/Contract modals generate txs

## Summary

Closes: https://github.com/snapshot-labs/sx-ui/issues/162

Changes:
- Modified `INumber` component. `modelValue` is what it needed to be (just implementation was breaking two-way binding, now we use new implementation, with old prop names, compatible with `v-model`).
- Added `ERC20`, `ERC721`, `ERC1155` ABIs.
- Added `Transaction` type to represent intended format for transactions.
- Enabled TS support in `SendToken`, `SendNft` and `ExecutionEditable` so we can verify that it conforms to the type. Skipped `Transaction` for not as it looks like it needs more changes and I don't want to pollute this PR even more.
- Made `SendToken`, `SendNft` and `Transaction` output tx in the same format.
- Updated `ExecutionEditable` to display those transactions types properly.

## Test plan
- Add send token execution (with ERC20 and ETH).
- Add send NFT execution.
- Add contract call execution.
- All of them should look in execution list as expected.
- Inspect generated `tx` objects that contents look good.

## Screenshots
<img src="https://user-images.githubusercontent.com/1968722/174681602-d1ece1d6-c64d-491a-94f7-a5dff6c96557.png" width="400" />

## Comments

Submitting modals might fail (for example amount is invalid or contract call is invalid for some reason). It would be nice to display some kind of error if that happens. I left it out of this PR as it's pretty big already.
